addons/mail/models/mail_thread.py

```diff
@@ -1409,6 +1409,9 @@ def _message_parse_extract_payload(self, message, save_original=False):
                     alternative = True
                 if part.get_content_type() == 'multipart/mixed':
                     mixed = True
+                if part.get_content_type() == 'bin/plain' and 'attachment' in part.get('Content-Disposition'):
+                    _logger.warning("Message containing an unexpected Content-Type, assuming 'attachment'")
+                    part.replace_header('Content-Type', 'application/pdf;' + part.get('Content-Type', '').split(';')[1])
```

Comment:
As we get the name, and "bin/plain" seems to imply more "octet-stream" than "pdf" to me by default, I would say we only convert to pdf if we find ".pdf" in the filename parameter, as it seems the problematic service at least gives us that. And/Or checking the charset isn't "binary"